examples/ASDKgram/Sample/PhotoFeedNodeController.m

```diff
@@ -0,0 +1,158 @@
+
+- (void)refreshFeed
+{
+  // small first batch
+  [_photoFeed refreshFeedWithCompletionBlock:^(NSArray *newPhotos){
+    
+    [self insertNewRowsInTableView:newPhotos];
+    [self requestCommentsForPhotos:newPhotos];
+    
+    // immediately start second larger fetch
+    [self loadPage];
+    
+  } numResultsToReturn:4];
+}
+
+- (void)loadPage
+{
+  [_photoFeed requestPageWithCompletionBlock:^(NSArray *newPhotos){
+    [self insertNewRowsInTableView:newPhotos];
+    [self requestCommentsForPhotos:newPhotos];
+  } numResultsToReturn:20];
+}
+
+- (void)requestCommentsForPhotos:(NSArray *)newPhotos
+{
+  for (PhotoModel *photo in newPhotos) {
+    [photo.commentFeed refreshFeedWithCompletionBlock:^(NSArray *newComments) {
+      
+      NSInteger rowNum      = [_photoFeed indexOfPhotoModel:photo];
+      NSIndexPath *cellPath = [NSIndexPath indexPathForRow:rowNum inSection:0];
+      PhotoCellNode *cell   = (PhotoCellNode *)[_tableNode.view nodeForRowAtIndexPath:cellPath];
+      
+      if (cell) {
+        [cell loadCommentsForPhoto:photo];
+        [_tableNode.view beginUpdates];
+        [_tableNode.view endUpdates];
+      }
+    }];
+  }
+}
+
+- (void)insertNewRowsInTableView:(NSArray *)newPhotos
+{
+  NSInteger section = 0;
+  NSMutableArray *indexPaths = [NSMutableArray array];
+  
+  NSUInteger newTotalNumberOfPhotos = [_photoFeed numberOfItemsInFeed];
+  for (NSUInteger row = newTotalNumberOfPhotos - newPhotos.count; row < newTotalNumberOfPhotos; row++) {
+    NSIndexPath *path = [NSIndexPath indexPathForRow:row inSection:section];
+    [indexPaths addObject:path];
+  }
+  
+  [_tableNode.view insertRowsAtIndexPaths:indexPaths withRowAnimation:UITableViewRowAnimationNone];
+}
+
+- (UIStatusBarStyle)preferredStatusBarStyle
+{
+  return UIStatusBarStyleLightContent;
+}
+
+- (CGSize)imageSizeForScreenWidth
+{
+  CGRect screenRect   = [[UIScreen mainScreen] bounds];
+  CGFloat screenScale = [[UIScreen mainScreen] scale];
+  return CGSizeMake(screenRect.size.width * screenScale, screenRect.size.width * screenScale);
+}
+
+#pragma mark - ASTableDataSource methods
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+  return [_photoFeed numberOfItemsInFeed];
+}
+
+- (ASCellNode *)tableView:(ASTableView *)tableView nodeForRowAtIndexPath:(NSIndexPath *)indexPath;
+{
+  PhotoCellNode *cellNode = [[PhotoCellNode alloc] initWithPhotoObject:[_photoFeed objectAtIndex:indexPath.row]];
+  
+  return cellNode;
+}
+
+#pragma mark - ASTableDelegate methods
+
+// table automatic tail loading
+-(void)scrollViewDidScroll:(UIScrollView *)scrollView
+{
+  CGFloat currentOffSetY = scrollView.contentOffset.y;
+  CGFloat contentHeight  = scrollView.contentSize.height;
+  CGFloat screenHeight   = [UIScreen mainScreen].bounds.size.height;
+  
+  CGFloat screenfullsBeforeBottom = (contentHeight - currentOffSetY) / screenHeight;
+  if (screenfullsBeforeBottom < AUTO_TAIL_LOADING_NUM_SCREENFULS) {
+    [self loadPage];
```

Comment:
This is cool, but the framework supports automatic batch loading - please adopt ASBatchContext